cylc/uiserver/handlers.py

```diff
@@ -148,6 +200,13 @@ def select_subprotocol(self, subprotocols):
         return GRAPHQL_WS
 
     @websockets_authenticated
+    @authorised
+    def get(self, *args, **kwargs):
+        # forward this call so we can authenticate/authorise it
+        return websocket.WebSocketHandler.get(self, *args, **kwargs)
+
+    @websockets_authenticated
+    @authorised
     def open(self, *args, **kwargs):
```

Comment:
I think this is never called without passing through `get` first. We will start an HTTP GET request with the [upgrade header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Protocol_upgrade_mechanism). Then once that's accepted and authenticated and authorized, then the connection opens.